eZ/Publish/Core/MVC/Symfony/SiteAccess/Matcher/HostElement.php

```diff
@@ -37,10 +37,15 @@ class HostElement implements VersatileMatcher
     /**
      * Constructor.
      *
-     * @param int $elementNumber Number of elements to take into account.
+     * @param int|array $elementNumber Number of elements to take into account.
      */
     public function __construct($elementNumber)
     {
+        if (is_array($elementNumber) && isset($elementNumber['value'])) {
```

Comment:
What Yannick would be best in semver terms:
- fix the behavior in 5.4.x / 1.0 (it's what users expect anyway)
- mark as deprecated in 1.1 (with notice)
- remove deprecated behaviour in future major release (7.0 kernel)

According to semver, the deprecation notice isn't supposed to be backported (